packages/vanilla/src/controls/RadioGroup.tsx

```diff
@@ -0,0 +1,116 @@
+/*
+  The MIT License
+
+  Copyright (c) 2017-2021 EclipseSource Munich
+  https://github.com/eclipsesource/jsonforms
+
+  Permission is hereby granted, free of charge, to any person obtaining a copy
+  of this software and associated documentation files (the "Software"), to deal
+  in the Software without restriction, including without limitation the rights
+  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+  copies of the Software, and to permit persons to whom the Software is
+  furnished to do so, subject to the following conditions:
+
+  The above copyright notice and this permission notice shall be included in
+  all copies or substantial portions of the Software.
+
+  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+  THE SOFTWARE.
+*/
+import React from 'react';
+import {
+  computeLabel,
+  ControlProps,
+  ControlState,
+  EnumOption,
+  enumToEnumOptionMapper,
+  isDescriptionHidden,
+  JsonSchema,
+  oneOfToEnumOptionMapper
+} from '@jsonforms/core';
+import { Control } from '@jsonforms/react';
+import { VanillaRendererProps } from '../index';
+import merge from 'lodash/merge';
+
+export class RadioGroup extends Control<
+  ControlProps & VanillaRendererProps,
+  ControlState
+> {
+  render() {
+    const {
+      classNames,
+      id,
+      label,
+      required,
+      description,
+      errors,
+      data,
+      schema,
+      uischema,
+      visible,
+      config
+    } = this.props;
+    const isValid = errors.length === 0;
+    const divClassNames = `validation  ${isValid ? classNames.description : 'validation_error'
+      }`;
+    const groupStyle: { [x: string]: any } = {
+      display: 'flex',
+      flexDirection: 'row'
+    };
+
+    const appliedUiSchemaOptions = merge({}, config, uischema.options);
+    const showDescription = !isDescriptionHidden(
+      visible,
+      description,
+      this.state.isFocused,
+      appliedUiSchemaOptions.showUnfocusedDescription
+    );
+
+    const options: EnumOption[] =
+      (schema.enum?.map(enumToEnumOptionMapper)) ||
+      (schema.oneOf &&
+        (schema.oneOf as JsonSchema[]).map(oneOfToEnumOptionMapper)
+      );
```

Comment:
This manual computation is not necessary, we offer specific bindings for this, i.e. `withJsonFormsEnumControlProps` and `withJsonFormsOneOfEnumProps`. The bindings eventually call [mapStateToEnumControlProps](https://github.com/eclipsesource/jsonforms/blob/ba6bdc5a93044a872285cd71ed40d9a54f437561/packages/core/src/util/renderer.ts#L457-L470) or [mapStateToOneOfEnumControlProps](https://github.com/eclipsesource/jsonforms/blob/ba6bdc5a93044a872285cd71ed40d9a54f437561/packages/core/src/util/renderer.ts#L478-L490) which do this mapping for you and will hand them over as `options` prop.
```suggestion
```